### PR TITLE
All the P

### PR DESCRIPTION
**Pull request by chosta**
Note that I've also removed the HTML_DATA for N, O to avoid conflicts in merging. I'll make sure N + O comes before P when merging, so don't worry about blank cards.

### CODE REVIEW

File: src/cards/base/Pets.ts [hunk 1]
```diff
@@ -28,4 +32,18 @@ export class Pets implements IProjectCard, IResourceCard {
...
     }
+    public metadata: CardMetadata = {
+      cardNumber: '172',
+      renderData: CardRenderer.builder((b) => {
+        b.effectBox((eb) => {
+          eb.city(CardRenderItemSize.SMALL).any.startEffect.animals(1);
+          eb.description('Effect: When any City tile is placed, add an Animal to this card.');
+        }).br;
+        b.animals(1).br;
+        b.text('Animals may not be removed from this card', CardRenderItemSize.SMALL, true).br;
```
Comment (kberg) [inline src/cards/base/Pets.ts:43]: Isn't this meant to be capitalized?

File: src/cards/base/PowerGrid.ts [hunk 2]
```diff
@@ -16,4 +17,12 @@ export class PowerGrid implements IProjectCard {
       player.addProduction(Resources.ENERGY, 1 + player.getTagCount(Tags.ENERGY));
       return undefined;
     }
+
+    public metadata: CardMetadata = {
+      cardNumber: '102',
+      renderData: CardRenderer.builder((b) => {
+        b.productionBox((pb) => pb.energy(1).slash().energy(1).played);
```
Comment (kberg) [inline src/cards/base/PowerGrid.ts:24]: Another case of resource vs tag, and here, the second energy actually ought to be Power. Your thoughts?

File: src/cards/turmoil/ParliamentHall.ts [hunk 3]
```diff
@@ -29,4 +32,16 @@ export class ParliamentHall implements IProjectCard {
...
+
+    public metadata: CardMetadata = {
+      cardNumber: 'T08',
+      requirements: CardRequirements.builder((b) => b.party(PartyName.MARS)),
+      renderData: CardRenderer.builder((b) => {
+        b.productionBox((pb) => {
+          pb.megacredits(1).slash().building(3).played;
+        });
+      }),
+      description: 'Requires that Mars First are ruling or that you have 2 delegates there. Increase your MC production 1 step for every 3 Building tag you have, including this.',
```
Comment (kberg) [inline src/cards/turmoil/ParliamentHall.ts:44]: "every 3 Building tag" should be "tags".

**Review by kberg — APPROVED** [review]
Looks great.

File: src/cards/base/Pets.ts (hunk 1 above)
Comment (chosta) [reply src/cards/base/Pets.ts:43]: It is, by the `true` parameter at the end. I wish TS had something like python's parameter naming
`text(isUppercase=True)`
In ES6 you can do `text({isUppercase=True})` but it doesn't appeal to me.

File: src/cards/base/PowerGrid.ts (hunk 2 above)
Comment (chosta) [reply src/cards/base/PowerGrid.ts:24]: It's on my refactoring list. Certainly better semantics could be used in these cases. 
E.g.
CardRequirements.builder((b) => b.tag(Tags.VENUS).tag(Tags.JOVIAN).tag(Tags.EARTH))
can become:
`CardRequirements.builder((b) => b.unity())` [From turmoil, unity is the bonus for jov, earth, venus tags] its conceptually appropriate and it is repeated multiple times.

**Pull request merged**